### PR TITLE
Add optional support for using getters in serialization

### PR DESCRIPTION
**Pull request by stashuk**
Adding a getter support at @JsonField class-level annotation for serialization, updating SerializerTest to include new functionality.

### CODE REVIEW

File: processor/src/main/java/com/instagram/common/json/annotation/processor/JsonParserClassData.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
   private final boolean mPostprocessingEnabled;
   private final String mValueExtractFormatter;
   private final String mParentInjectedClassName;
+  private final boolean mUseGetters;
 
   public JsonParserClassData(
```
Comment (ttung) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonParserClassData.java]: at this point, this class is a total mess.  someone who comes along to touch this code is at high risk of breaking something.  a ton of this information is coming straight from TypeData.  can this be refactored to accept an instance of TypeData?

Comment (ttung) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonParserClassData.java]: does it make sense just to pass the class annotation into `JsonParserClassData`?

File: processor/src/test/java/com/instagram/common/json/annotation/processor/SerializeTest.java [hunk 2]
```diff
@@ -333,4 +333,23 @@ public void serializeOrderTest() throws IOException {
               < inputString.indexOf("\"" + FIELD_DECLARATION_ORDER[i + 1] + "\""));
     }
   }
+
+  @Test
+  public void serializeWithGetterTest() throws IOException {
+    SimpleParseUUT source = new SimpleParseUUT();
```
Comment (ttung) [inline processor/src/test/java/com/instagram/common/json/annotation/processor/SerializeTest.java]: don't overload `SimpleParseUUT`.  this is not basic functionality.

**Comment by stashuk** [comment]
Agree, using GetterUUT now and passing annotation directly.

**Comment by ttung** [comment]
this looks good, but the rev graph is confusing.  can you make one commit that adds the getter feature, and one commit that cleans up the code.  either order is fine.  :)

**Comment by stashuk** [comment]
Awesome, thanks! Changing to two commits now.

**Comment by stashuk** [comment]
Ping on merge.

**Pull request merged**